service/javadsl/integration-tests/src/test/java/com/lightbend/lagom/it/routers/AdditionalRoutersServiceModule.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.it.routers;
+
+import com.google.inject.AbstractModule;
+import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport;
+
+public class AdditionalRoutersServiceModule  extends AbstractModule implements ServiceGuiceSupport {
+    @Override
+    protected void configure() {
+        bindService(
+            AdditionalRoutersService.class, AdditionalRoutersServiceImpl.class,
+
+            // bind a router using an instance and prefix it using the bind dsl
+            additionalRouter(PingRouter.newInstance()).withPrefix("/ping"),
+
+            // bind a prefixed router using an instance
+            additionalRouter(PongRouter.newInstance()),
+
+            // bind a router using a class (DI case) and prefix it using bind dsl
+            additionalRouter(HelloRouter.class).withPrefix("/hello"),
+
+            // bind a prefixed router using a class (DI case)
+            // (that router is already prefixed with /hello-prefixed)
+            additionalRouter(PrefixedHelloRouter.class)
+        );
+    }
```

Comment:
This is the final API I came up after experimenting with different options. 

This allows for configuration of `Routers` requiring DI and simple `Router` instances. I had to introduce an indirection to make it work. We have a new class, `AdditionalRouter`, that works as a wrapper/builder that is then used later to wire the `Router` and attach it to the default Lagom router.

Reply:
I'm a little conflicted about this. I like that it's a little more type-safe and explicit, but it seems non-idiomatic for how Play normally handles configurable dynamic runtime injection, which is to load classes from configuration.

https://github.com/playframework/playframework/blob/302b2506fee044e8e141abc8654f50d0b069e80d/framework/src/play/src/main/scala/play/api/inject/BuiltinModule.scala#L86-L92

For example, the list of filters is analogous: https://www.playframework.com/documentation/2.6.x/Filters#Default-Filters

You're going to hate me, but it makes me wonder whether we'd be better off allowing a configurable list of routers upstream in Play, rather than adding it as a Lagom-specific thing.

@marcospereira what do you think?

Reply:
Upstream in Play, the 'configurable list of routers' is basically the `routes.txt`. Adding gRPC routes in there rather than a new configuration option in the `.conf` seemed consistent. Since Lagom doesn't use the `routes.txt` but instead builds the service in code, it sort of made sense to me to add the additional routers in code as well. But that is without much Play/Lagom experience, of course :).

Reply:
Let's just go with this for now, and then see how it feels as we start building out examples.